patches/chromium/process_singleton.patch

```diff
@@ -23,8 +23,35 @@ This patch adds a few changes to the Chromium code:
    `ProcessSingleton` instance that tries to connect to the socket
    before the browser thread is ready.
 
+diff --git a/chrome/browser/chrome_browser_main.cc b/chrome/browser/chrome_browser_main.cc
+index aca2f19f53657fc2b944b69418e099b056c8e734..e43b0d89e2e6ad0bcb2c34bfacc4ef3b59eaaab1 100644
+--- a/chrome/browser/chrome_browser_main.cc
++++ b/chrome/browser/chrome_browser_main.cc
+@@ -1424,7 +1424,6 @@ int ChromeBrowserMainParts::PreMainMessageLoopRunImpl() {
+   switch (notify_result_) {
+     case ProcessSingleton::PROCESS_NONE:
+       // No process already running, fall through to starting a new one.
+-      process_singleton_->StartWatching();
```

Comment:
why are we patching `chrome_browser_main.cc`? Surely we don't run this code?

Reply:
I haven't tested re-adding this to see if this original rationale still applies, but according to the patch the reason for this change is:

> 4. It adds an `OnBrowserReady` function to allow
>   `requestSingleInstanceLock` to start listening to the socket later
>   in the posix implementation. This function is necessary, because
>   otherwise, users calling `requestSingleInstanceLock` create a
>   `ProcessSingleton` instance that tries to connect to the socket
>   before the browser thread is ready.

Reply:
I'm pretty sure `ChromeBrowserMainParts` is never instantiated or referenced in our code, so this patch is a no-op.

Reply:
The chrome side of changes are not required, our call to `ProcessSingleton` instantiation happens on demand from https://github.com/electron/electron/blob/main/shell/browser/api/electron_api_app.cc#L1126-L1136 . Also, I think https://chromium-review.googlesource.com/c/chromium/src/+/3781028 works in our favor to delay the watcher which aligns with point 4). I will try to fix this patch today.